Podcast/LoginViewController.swift

```diff
@@ -0,0 +1,131 @@
+            make.height.equalToSuperview().multipliedBy(0.25)
+            make.top.equalTo(0).inset(view.frame.width * 0.25)
+        }
+
+        facebookLoginButton = UIButton()
+        facebookLoginButton.setBackgroundImage(#imageLiteral(resourceName: "signinFb"), for: .normal)
+        facebookLoginButton.addTarget(self, action: #selector(facebookLoginButtonPress), for: .touchUpInside)
+        view.addSubview(facebookLoginButton)
+
+        facebookLoginButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(podcastLogoView.snp.bottom).offset(signInButtonTopPadding)
+            make.width.equalTo(signInButtonWidth)
+            make.height.equalTo(signInButtonHeight)
+        }
+
+        googleLoginButton = UIButton()
+        googleLoginButton.setBackgroundImage(#imageLiteral(resourceName: "signinGoogle"), for: .normal)
+        googleLoginButton.addTarget(self, action: #selector(googleLoginButtonPress), for: .touchUpInside)
+        view.addSubview(googleLoginButton)
+
+        googleLoginButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(facebookLoginButton.snp.bottom).offset(signInButtonSmallPadding)
+            make.width.equalTo(signInButtonWidth)
+            make.height.equalTo(signInButtonHeight)
+        }
+
+        loadingActivityIndicator = LoadingAnimatorUtilities.createLoadingAnimator()
+        loadingActivityIndicator.center = view.center
+        loadingActivityIndicator.color = .offWhite
+        loadingActivityIndicator.startAnimating()
+        view.addSubview(loadingActivityIndicator)
+        
+        hideLoginButtons(isHidden: true)
+
+        // if we have a valid access token for Facebook or Google then sign in silently
+        if let _ = Authentication.sharedInstance.facebookAccessToken {
+            // try signing in with Facebook
+            print(Authentication.sharedInstance.facebookAccessToken)
+            Authentication.sharedInstance.authenticateUser(signInType: .Facebook, success: self.signInSuccess, failure: self.signInFailure)
+        } else {
+            Authentication.sharedInstance.signInSilentlyWithGoogle() // Google delegate method will be called when this completes
+        }
+    }
+
+    @objc func googleLoginButtonPress() {
+        hideLoginButtons(isHidden: true)
+        loadingActivityIndicator.startAnimating()
+        Authentication.sharedInstance.signInWithGoogle()
+    }
+
+    @objc func facebookLoginButtonPress() {
+        hideLoginButtons(isHidden: true)
+        loadingActivityIndicator.startAnimating()
+        Authentication.sharedInstance.signInWithFacebook(viewController: self, success: {
+            Authentication.sharedInstance.authenticateUser(signInType: .Facebook, success: self.signInSuccess, failure: self.signInFailure)
+        }, failure: self.signInFailure)
+    }
+
+    func signInWithGoogle(wasSuccessful: Bool) {
```

Comment:
Okay so the one in `Authentication.sharedInstance` is the delegate Google requires you to have when the login is complete .. when Google login is complete it calls the delegate method in `Authentication.sharedInstance`. That method calls the `LogingViewController.signInWithGoogle` which then passes in failure and success blocks to the authenticated user request. We need both because the Google delegate method does not allow you to pass in completion blocks